src/AWS.Deploy.Common/Recipes/Validation/FargateTaskSizeCpuMemoryLimitsRecipeValidator.cs

```diff
@@ -0,0 +1,92 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.\r
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace AWS.Deploy.Common.Recipes.Validation
+{
+    /// <summary>
+    /// Enforces Fargate cpu and memory conditional requirements:
+    /// <para/> CPU value  Memory value (MiB)
+    /// <para/> 256 (.25 vCPU)  512 (0.5 GB), 1024 (1 GB), 2048 (2 GB)
+    /// <para/> 512 (.5 vCPU)   1024 (1 GB), 2048 (2 GB), 3072 (3 GB), 4096 (4 GB)
+    /// <para/> 1024 (1 vCPU)   2048 (2 GB), 3072 (3 GB), 4096 (4 GB), 5120 (5 GB), 6144 (6 GB), 7168 (7 GB), 8192 (8 GB) 
+    /// <para/> 2048 (2 vCPU)   Between 4096 (4 GB) and 16384 (16 GB) in increments of 1024 (1 GB) 
+    /// <para/> 4096 (4 vCPU)   Between 8192 (8 GB) and 30720 (30 GB) in increments of 1024 (1 GB)
+    /// <para />
+    /// See https://docs.aws.amazon.com/AmazonECS/latest/userguide/task_definition_parameters.html#task_size
+    /// for more details.
+    /// </summary>
+    public class FargateTaskSizeCpuMemoryLimitsRecipeValidator : IRecipeValidator
```

Comment:
nit: I suggest simplifying the name. change it to "FargateTaskCpuMemorySizeValidator". Also, separate the recipe validators and option setting validators. Have them in different folders which would have the type in the namespace.

Reply:
Changed the name and moved the validators in separate folders. However, assigning them separate namespaces is not required.